pickhardtpayments/SyncSimulatedPaymentSession.py

```diff
@@ -333,57 +344,85 @@ def activate_network_wide_uncertainty_reduction(self, n):
-        full_amt = amt
         cnt = 0
         total_fees = 0
-        number_number_of_onions = 0
         total_number_failed_paths = 0
 
+        # Initialise Payment
+        # currently with underscore to not mix up with existing variable 'payment'
+        payment = Payment(src, dest, amt)
+
         # This is the main payment loop. It is currently blocking and synchronous but may be
-        # implemented in a concurrent way. Also we stop after 10 rounds which is pretty arbitrary
-        # a better stop criteria would be if we compute infeasable flows or if the probabilities
-        # are to low or residual amounts decrease to slowly
+        # implemented in a concurrent way. Also, we stop after 10 rounds which is pretty arbitrary
+        # a better stop criteria would be if we compute infeasible flows or if the probabilities
+        # are too low or residual amounts decrease to slowly
         while amt > 0 and cnt < 10:
-            print("Round number: ", cnt+1)
+            print("Round number: ", cnt + 1)
             print("Try to deliver", amt, "satoshi:")
 
-            # transfer to a min cost flow problem and rund the solver
-            paths, runtime = self._generate_candidate_paths(
-                src, dest, amt, mu, base)
+            # transfer to a min cost flow problem and run the solver
+            # paths is the lists of channels, runtime the time it took to calculate all candidates in this round
+            attempts_in_round, runtime = self._generate_candidate_paths(
+                payment.sender, payment.receiver, amt, mu, base)
 
             # compute some statistics about candidate paths
-            payments = self._estimate_payment_statistics(paths)
+            self._estimate_payment_statistics(attempts_in_round)
 
-            # matke attempts and update our information about the UncertaintyNetwork
-            self._attempt_payments(payments)
+            # make attempts, try to send onion and register if success or not
+            # update our information about the UncertaintyNetwork
+            self._attempt_payments(attempts_in_round)
+
+            # add attempts to payment
+            payment.add_attempts(attempts_in_round)
 
             # run some simple statistics and depict them
             amt, paid_fees, num_paths, number_failed_paths = self._evaluate_attempts(
-                payments)
+                attempts_in_round)
+
             print("Runtime of flow computation: {:4.2f} sec ".format(runtime))
             print("\n================================================================\n")
 
-            number_number_of_onions += num_paths
             total_number_failed_paths += number_failed_paths
             total_fees += paid_fees
             cnt += 1
-        end = time.time()
+        # When residual amount is 0 / enough successful onions have been found, then settle payment. Else drop onions.
+        if amt == 0:
+            for onion in payment.inflight_attempts:
```

Comment:
technically inflight might not be at recipient yet and could still fail. So I guess we want `ARRIVED` here. I guess that was the confusion of this artificial distinction between `INFLIGHT` and `ARRIVED`

Reply:
Yes. I think that indeed it is a lot clearer with ARRIVED.